app/models/host/managed.rb

```diff
@@ -926,6 +927,17 @@ def apply_compute_profile(modification)
 
   private
 
+  def owner_taxonomies_match
+    if self.owner.is_a?(User)
```

Comment:
Move this `if` to the `validate` above, no point on going in the method if this condition is not met